test/sql/insert_sql_test.cpp

```diff
@@ -570,5 +567,26 @@ TEST_F(InsertSQLTests, UniqueColumn) {
   txn_manager.CommitTransaction(txn);
 }
 
+TEST_F(InsertSQLTests, NonExistentTable) {
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->CreateDatabase(DEFAULT_DB_NAME, txn);
+  txn_manager.CommitTransaction(txn);
+  std::string error_message;
+  std::unique_ptr<optimizer::AbstractOptimizer> optimizer(
+      new optimizer::Optimizer());
+  // Insert an int into a non-existent table.
+  std::string query("INSERT INTO NonExistentTable VALUES(3);");
+  txn = txn_manager.BeginTransaction();
+  EXPECT_THROW({
+    try {
+      TestingSQLUtil::GeneratePlanWithOptimizer(optimizer, query, txn);
+    } catch (peloton::CatalogException &ex) {
+      EXPECT_STREQ("Table nonexistenttable is not found", ex.what());
+      throw peloton::CatalogException(ex.what());
```

Comment:
Just rethrow here with `throw;`

I'm still not a fan of string equality checks in tests. It creates a tight coupling and because in the long run they'll cause problems if/when we change the string format.

Can you simplify the check here to just fail if you get past the call to `TestingSQLUtil::GeneratePlanWithOptimizer`? You can remove the outer `EXPECT_THROW` .

Reply:
But if I'll rethrow, the test suite will report that as an error, right? If that's fine, I'll remove the outer `EXCEPT_THROW`. Otherwise, I can just do something like `EXCEPT_THROW(TestingSQLUtil::GeneratePlanWithOptimizer(optimizer, query, txn), peloton::CatalogException);`, just like what you suggested sometime earlier.

Reply:
Remove the outer throw, just fail if you get past the call to `GeneratePlanWithOptimizer()`.